### ISSUE TITLE
Import error when using pyglet

### ISSUE THREAD

**Issue by ashes999** [pyinstaller/pyinstaller#2296]
I'm trying to use PyInstaller on something build with pyglet. The final binary crashes with an `ImportError`.

Here's the script file I'm using (`main.py`):

`import pyglet`

Here's the output of `pyinstaller main.py`:

```
81 INFO: PyInstaller: 3.2
81 INFO: Python: 2.7.6
81 INFO: Platform: Linux-3.13.0-98-generic-x86_64-with-Ubuntu-14.04-trusty
81 INFO: wrote /home/ashiq/Dropbox/Python/pyinstall-test/main.spec
85 INFO: UPX is not available.
86 INFO: Extending PYTHONPATH with paths
['/home/ashiq/Dropbox/Python/pyinstall-test',
 '/home/ashiq/Dropbox/Python/pyinstall-test']
86 INFO: checking Analysis
86 INFO: Building Analysis because out00-Analysis.toc is non existent
86 INFO: Initializing module dependency graph...
106 INFO: Initializing module graph hooks...
163 INFO: running Analysis out00-Analysis.toc
461 INFO: Caching module hooks...
463 INFO: Analyzing /home/ashiq/Dropbox/Python/pyinstall-test/main.py
465 INFO: Loading module hooks...
465 INFO: Loading module hook "hook-encodings.py"...
5835 INFO: Looking for ctypes DLLs
5835 INFO: Analyzing run-time hooks ...
5841 INFO: Looking for dynamic libraries
6371 INFO: Looking for eggs
6371 INFO: Python library not in binary depedencies. Doing additional searching...
6953 INFO: Using Python library /usr/lib/x86_64-linux-gnu/libpython2.7.so.1.0
6958 INFO: Warnings written to /home/ashiq/Dropbox/Python/pyinstall-test/build/main/warnmain.txt
7058 INFO: checking PYZ
7058 INFO: Building PYZ because out00-PYZ.toc is non existent
7058 INFO: Building PYZ (ZlibArchive) /home/ashiq/Dropbox/Python/pyinstall-test/build/main/out00-PYZ.pyz
7334 INFO: checking PKG
7334 INFO: Building PKG because out00-PKG.toc is non existent
7334 INFO: Building PKG (CArchive) out00-PKG.pkg
7433 INFO: Bootloader /usr/local/lib/python2.7/dist-packages/PyInstaller/bootloader/Linux-64bit/run
7433 INFO: checking EXE
7433 INFO: Building EXE because out00-EXE.toc is non existent
7434 INFO: Building EXE from out00-EXE.toc
7480 INFO: Appending archive to ELF section in EXE /home/ashiq/Dropbox/Python/pyinstall-test/build/main/main
7717 INFO: checking COLLECT
7718 INFO: Building COLLECT because out00-COLLECT.toc is non existent
7718 INFO: Building COLLECT out00-COLLECT.toc
```

Here's the trackback when I run `main` from `dist`:

```
Traceback (most recent call last):
  File "main.py", line 1, in <module>
ImportError: No module named pyglet
Failed to execute script main
```

I tried using a hidden-import (both from the cmdline and from the .spec file), to no avail.

 `warnmain.txt` has 121 `missing module` lines. Sure enough, the last one is:

`missing module named pyglet - imported by /home/ashiq/Dropbox/Python/pyinstall-test/main.py`

I'm new to Python, Pyglet, and PyInstaller, so I don't know how to troubleshoot this further.

Info from the How to Report Bugs page:
- Using Python 2.7.6 (default for `python` on my OS), running the game with `python3`
- Using PyInstaller 3.2 (latest binaries available today)
- Using pyglet 1.2.4 (reported by pip)
- OS: Ubuntu 14.04 LTS

This seems to work with the Windows version of PyInstaller (2.1 and 3.2).

**Comment by ashes999** [comment]
Now I'm even more confused.  After building it successfully on Windows, I can build it successfully on Linux.  This is even if I delete `build`, `dist`, and `main.spec`.

I don't know what happened. Nothing changed on my Unix system.

**Comment by ashes999** [comment]
I think I figured this out.

While writing up this issue, I reached the question of what version of Python I'm using. While I run my game with `python3`, it's possible that PyInstaller picked up and packaged (if that's how it works?) the local version of Python 2 on my system, which is the default. (Running `python` lists `2.7.6` as the main version.)

I tried installing pyglet with `pip` instead of `pip3`. I'm pretty sure I re-built my app with PyInstaller and it still reported that `pyglet` is missing.

But, this could explain why it didn't work.  

I'm going to close this issue for now, since it consistently works.  In summary:
- Installed `pyglet` via `pip3`, built and ran game with `python3`
- Invoked `pyinstaller`, ran binary, it complained `pyglet` isn't found
- Installed `pyglet` via `pip`
- Invoked `pyinstaller`, ran binary, it worked
